vagrant/ansible/roles/samba.setup/tasks/main.yml

```diff
@@ -43,6 +46,11 @@
         insertafter: EOF
       with_items: "{{ samba_shares }}"
 
+- name: Restart samba
+  service:
+    name: smb
+    state: restarted
+
```

Comment:
I am not sure about a direct restart of `smb` service.

@obnoxxx @gd With CTDB already configured and running, shouldn't we try restarting `ctdb` service instead? IIRC, restarting `ctdb` should automatically take care of restarting `smb` too.

Reply:
@anoopcs9  wrote:
> I am not sure about a direct restart of `smb` service.
> 
> @obnoxxx @gd With CTDB already configured and running, shouldn't we try restarting `ctdb` service instead? IIRC, restarting `ctdb` should automatically take care of restarting `smb` too.

if `CTDB_MANAGES_SAMBA` is enabled (at least that was the name of the setting several years ago -- it might have changed...), then  ctdb will manage the smb service and will make sure it gets started. But  it should also be fine to manually (re)start the smb service. At least best I remember, neither ctdb nor samba in clustering mode  require samba to be managed by ctdb. the crucial piece is that both are running and can communicate.

Reply:
> @anoopcs9 wrote:
> 
> > I am not sure about a direct restart of `smb` service.
> > @obnoxxx @gd With CTDB already configured and running, shouldn't we try restarting `ctdb` service instead? IIRC, restarting `ctdb` should automatically take care of restarting `smb` too.
> 
> if `CTDB_MANAGES_SAMBA` is enabled (at least that was the name of the setting several years ago -- it might have changed...), then ctdb will manage the smb service and will make sure it gets started.

This option was later changed when CTDB got rid of event scripts for services with [v4.9.0](https://www.samba.org/samba/history/samba-4.9.0.html) and we currently enable those via the new sub commands to `ctdb` utility.

> But it should also be fine to manually (re)start the smb service. At least best I remember, neither ctdb nor samba in clustering mode require samba to be managed by ctdb. the crucial piece is that both are running and can communicate.

Agreed.